framework/audio/internal/sequencer.cpp

```diff
@@ -0,0 +1,262 @@
+}
+
+std::optional<Sequencer::track_t> Sequencer::track(Sequencer::track_id id) const
+{
+    if (m_tracks.find(id) != m_tracks.end()) {
+        return m_tracks.at(id);
+    }
+    return std::nullopt;
+}
+
+Sequencer::midi_track_t Sequencer::midiTrack(Sequencer::track_id id) const
+{
+    if (auto track_var = track(id)) {
+        if (auto track_value = track_var.value_or(nullptr)) {
+            return std::dynamic_pointer_cast<IMIDIPlayer>(track_value);
+        }
+    }
+    return nullptr;
+}
+
+Sequencer::midi_track_t Sequencer::createMIDITrack(ISequencer::track_id id)
+{
+    auto player = std::make_shared<MIDIPlayer>();
+    m_tracks[id] = player;
+    return player;
+}
+
+Sequencer::audio_track_t Sequencer::createAudioTrack(Sequencer::track_id id)
+{
+    auto player = std::make_shared<AudioPlayer>();
+    m_tracks[id] = player;
+    m_audioTrackAdded.send(player);
+    return player;
+}
+
+Sequencer::audio_track_t Sequencer::audioTrack(Sequencer::track_id id) const
+{
+    if (auto track_var = track(id)) {
+        if (auto track_value = track_var.value_or(nullptr)) {
+            return std::dynamic_pointer_cast<IAudioPlayer>(track_value);
+        }
+    }
+    return nullptr;
+}
+
+std::shared_ptr<IMIDIPlayer> Sequencer::playMidi(const midi::MidiData& data)
+{
+    auto player = std::make_shared<MIDIPlayer>();
+    auto task = [data, player]() {
+                    auto midiStream = std::make_shared<midi::MidiStream>();
+                    midiStream->initData = data;
+                    midiStream->isStreamingAllowed = false;
+                    midiStream->lastTick = data.lastChunksTick();
+
+                    auto clock = std::make_shared<Clock>();
+                    clock->setSampleRate(1000);
+                    clock->start();
+
+                    player->loadMIDI(midiStream);
+                    player->play();
+
+                    while (player->isRunning()) {
+                        clock->forward(50);
+                        player->forwardTime(clock->timeInMiliSeconds());
+                        std::this_thread::sleep_for(std::chrono::milliseconds(50));
+                    }
+                };
+    auto t = std::thread(task);
+    t.detach();
+    return player;
+}
+
+void Sequencer::buildRpcReflection()
+{
+    using namespace rpc;
+    m_rpcReflection = {
+        { "play",    [this](ArgumentList) -> Variable { return play(); } },
+        { "pause",   [this](ArgumentList) -> Variable { pause(); return {}; } },
+        { "stop",    [this](ArgumentList) -> Variable { stop(); return {}; } },
+        { "seek",    [this](ArgumentList args) -> Variable { seek(args[0].toUInt()); return {}; } },
+        { "rewind",  [this](ArgumentList)      -> Variable { rewind(); return {}; } },
+        { "setLoop", [this](ArgumentList args) -> Variable { setLoop(args[0].toUInt(), args[1].toUInt()); return {}; } },
+        { "unsetLoop",  [this](ArgumentList)   -> Variable { unsetLoop(); return {}; } }
+    };
```

Comment:
I think this binding should be outside the class, the class should not know anything about the RPC implementation (its types, signatures, etc.)

Reply:
fixed here https://github.com/musescore/MuseScore/pull/7266